src/components/Table/RowState.ts

```diff
@@ -80,23 +80,26 @@ export class RowState {
   }
 
   loadSelected(rows: GridDataRow<any>[]): void {
-    const selectedRows = rows.filter(row => row.initSelected)
-    // Initialize with selected rows as defined 
+    const selectedRows = rows.filter((row) => row.initSelected);
+    // Initialize with selected rows as defined
     const map = new Map<string, SelectedState>();
-    selectedRows.forEach(row => {
+    selectedRows.forEach((row) => {
       map.set(row.id, "checked");
-    })
+    });
 
     this.selectedRows.merge(map);
   }
 
   // Updates the list of rows and regenerates the collapsedRows property if needed.
   setRows(rows: GridDataRow<any>[]): void {
-    // If the set of rows are different, and this is the first time adding "data" rows (non "totals" or "header" rows), then collapsedRows may need to be updated.
-    if (rows !== this.rows && this.rows.some((r) => r.kind !== "totals" && r.kind !== "header")) {
+    // If the set of rows are different
+    if (rows !== this.rows) {
       const currentCollapsedIds = this.collapsedIds;
       // Create a list of the (maybe) new rows that should be initially collapsed
       const maybeNewCollapsedRows = flattenRows(rows).filter((r) => r.initCollapsed);
+      // Check against local storage for collapsed state only if this is the first render of "data" (non-header or totals) rows.
+      const checkLocalStorage =
+        this.persistCollapse && !this.rows.some((r) => r.kind !== "totals" && r.kind !== "header");
```

Comment:
Thought: I was surprised that `persistCollapse` has not already overwritten local storage; i.e. `RowState.loadCollapse` calls `sessionStorage.setItem`, but looks like that's only if there are non-zero collapsed rows, and the header/totals wouldn't be collapsed...